server.py

```diff
@@ -110,10 +93,14 @@ def log_out_user():
 def task_list():
     """Show list of tasks."""  # All task objects for this user
 
-    user_id = session['user']  # get the user_id from the session dictionary
-    tasks = Task.query.filter_by(user_id=user_id).all()  # get list of tasks
+    user_id = session.get('user')
+    if user_id:
+        tasks = Task.query.filter_by(user_id=user_id).all()  # get list of tasks
```

Comment:
You can also do: `tasks = User.query.get(user_id).tasks`, especially nice if this view is for the given user.